portal/server/routes/middleware/validateRole/index.js

```diff
@@ -21,14 +21,15 @@ const userRoleIsValid = (requiredRoles, user) => {
  * @param {object} opts  (i.e. { role: ['maker'] })
  *
  */
-const validateRole = (opts) => {
+const validateRole = (opts, getRedirectUrl) => {
   const requiredRoles = opts ? opts.role : null;
 
   return (req, res, next) => {
     if (userRoleIsValid(requiredRoles, req.session.user)) {
       next();
     } else {
-      res.redirect('/');
+      const redirectUrl = getRedirectUrl ? getRedirectUrl(req) : '/';
+      res.redirect(redirectUrl);
     }
```

Comment:
Might be slightly nicer to make `() => '/'` the default for the `getRedirectUrl` argument so you can see what happens if you don't specify it from the first line of the function, instead of having to dive into the function body

Reply:
Nice, I like this